javascript/lifecycle.js

```diff
@@ -125,17 +126,50 @@ document.addEventListener(
 // - element - the element that triggered the reflex (not necessarily the Stimulus controller's element)
 //
 export const dispatchLifecycleEvent = (stage, element, reflexId) => {
-  if (!element) return
-  if (!element.reflexData) element.reflexData = {}
-  if (!element.reflexController) element.reflexController = {}
+  if (!element) {
+    if (Debug.enabled)
+      console.warn(
+        `StimulusReflex was not able execute the "${stage}" lifecycle method on the element which triggered the Reflex. The element is no longer present in the DOM. Could you move the Reflex controller to an element higher in your DOM?`
+      )
+    return
+  }
+
+  const reflexData = element.reflexData || {}
+  const reflexController = element.reflexController || {}
+  const oldElement = element
+
+  if (!document.body.contains(element)) {
+    const attrs = extractElementAttributes(element)
+    element = findElement(attrs)
+
+    if (Debug.enabled)
+      console.warn(
+        `StimulusReflex detected that the element which triggered the Reflex has been replaced by a morph operartion. If you rely on all lifecycle methods to be executed, move the Reflex controller to an element higher in your DOM.`
+      )
+  }
+
+  if (!element) {
+    if (Debug.enabled)
+      console.warn(
+        `StimulusReflex was not able execute the "${stage}" lifecycle method on the element which triggered the Reflex. The following element is no longer present in the DOM: `,
+        oldElement
+      )
+    return
+  }
+
+  element.reflexData = reflexData
```

Comment:
I'm a little bit concerned about these...

I think they should be:
```js
element.reflexData[reflexId] = reflexData
element.reflexController[reflexId] = reflexController
```

Reply:
I think this should be fine, because we are setting of these variables with:
```js
  const reflexData = element.reflexData || {}
  const reflexController = element.reflexController || {}
```